src/main/java/jewellery/inventory/mapper/SaleMapper.java

```diff
@@ -0,0 +1,128 @@
+    saleResponseDto.setProducts(mapAllProductsToResponse(sale));
+    saleResponseDto.setTotalPrice(getTotalPriceFromEntity(sale.getProducts()));
+    saleResponseDto.setTotalDiscount(getTotalDiscountPercentageFromEntity(sale.getProducts()));
+    saleResponseDto.setTotalDiscountedPrice(getTotalDiscountFromEntity(sale.getProducts()));
+    return saleResponseDto;
+  }
+
+  public Sale mapRequestToEntity(SaleRequestDto saleRequestDto) {
+    Sale sale = new Sale();
+    sale.setBuyer(getUserFromSaleRequestDto(saleRequestDto.getBuyerId()));
+    sale.setSeller(getUserFromSaleRequestDto(saleRequestDto.getSellerId()));
+    sale.setProducts(getProductsFromSaleRequestDto(saleRequestDto));
+    sale.setDate(Date.from(Instant.now()));
+    sale.setDiscount(getDiscountInPercentage(saleRequestDto.getProducts()));
+    return sale;
+  }
+
+  private List<ProductResponseDto> mapAllProductsToResponse(Sale sale) {
+    List<ProductResponseDto> productResponseDtos = new ArrayList<>();
+    for (Product product : sale.getProducts()) {
+      ProductResponseDto productResponseDto = productMapper.mapToProductResponseDto(product);
+      productResponseDtos.add(productResponseDto);
+    }
+    return productResponseDtos;
+  }
+
+  private List<Product> getProductsFromSaleRequestDto(SaleRequestDto saleRequestDto) {
+    List<Product> productList =
+        saleRequestDto.getProducts().stream()
+            .map(
+                productPriceDiscountRequestDto ->
+                    productRepository
+                        .findById(productPriceDiscountRequestDto.getProductId())
+                        .orElseThrow(
+                            () ->
+                                new ProductNotFoundException(
+                                    productPriceDiscountRequestDto.getProductId())))
+            .toList();
+    return setProductPriceAndDiscount(saleRequestDto, productList);
+  }
+
+  private double getDiscountInPercentage(List<ProductPriceDiscountRequestDto> discountList) {
+    double discountPercentage = 0;
+    for (ProductPriceDiscountRequestDto productPriceDiscountRequestDto : discountList) {
+      discountPercentage += productPriceDiscountRequestDto.getDiscount();
+    }
+    return discountPercentage / discountList.size();
+  }
+
+  private double getTotalPriceFromEntity(List<Product> products) {
+    double totalPrice = 0;
+    for (Product product : products) {
+      totalPrice += product.getSalePrice();
+    }
+    return totalPrice;
+  }
+
+  private double getTotalDiscountPercentageFromEntity(List<Product> products) {
+    double totalDiscountAmount = 0;
+    double totalPrice = 0;
+
+    for (Product product : products) {
+      double discountAmount = product.getSalePrice() * (product.getDiscount() / 100);
+      totalDiscountAmount += discountAmount;
+      totalPrice += product.getSalePrice();
+    }
+    if (totalPrice != 0) {
+      return (totalDiscountAmount / totalPrice) * 100;
```

Comment:
This would be the percentage value of the discountedAmount from the totalPrice. To get the discount percentage you should do
```suggestion
      return 100 - (totalDiscountAmount / totalPrice) * 100;
```